avaframe/com1DFAPy/com1DFA.py

```diff
@@ -55,6 +53,187 @@
 rng = np.random.default_rng(seed)
 
 
+def com1DFAMain(cfg, avaDir, relTh):
+    """ Run main model
+
+    This will compute a dense flow avalanche
+
+    Parameters
+    ----------
+    cfg : dict
+        configuration read from ini file
+    avaDir : str
+        path to avalanche directory
+
+    Returns
+    -------
+    reportDictList : list
+        list of dictionaries that contain information on simulations that can be used for report generation
+    """
+
+    # Setup configuration
+    cfgGen = cfg['GENERAL']
+    modName = 'com1DFAPy'
+
+    # Log current avalanche directory
+    log.debug('Your current avalanche name: %s' % avaDir)
+
+    # Create output and work directories
+    workDir, outDir = inDirs.initialiseRunDirs(avaDir, modName)
+
+    # Load input data
+    flagDev = cfg['FLAGS'].getboolean('flagDev')
+    avalancheDir = cfgGen['avalancheDir']
+    # fetch input data - dem, release-, entrainment- and resistance areas
+    demFile, relFiles, entFiles, resFile, EntRes = gI.getInputData(
+        avalancheDir, cfg['FLAGS'], flagDev)
+    # save flags
+    cfg['FLAGS']['EntRes'] = str(EntRes)
+    demOri, releaseLine, entLine, resLine, entrainmentArea, resistanceArea = prepareInputData(demFile, relFiles, entFiles, resFile)
+
+    # Counter for release area loop
+    countRel = 0
+
+    # Setup simulation dictionaries for report genereation
+    reportDictList = []
+
+    # Loop through release areas
+    for rel in relFiles:
+        # find out which simulations to perform
+        relName, cuSim, relDict, BadName = getSimulation(cfg['FLAGS'], rel)
+        for sim in cuSim:
+            if '_entres_' in sim:
+                if entLine:
+                    log.info('Simulation is performed using entrainment: %s' % (entLine['Name']))
+                if resLine:
+                    log.info('Simulation is performed using resistance: %s' % (resLine['Name']))
+            else:
+                log.info('Standard simulation is performed without entrainment and resistance')
+            logName = sim + '_' + cfgGen['mu']
+
+            # If initialisation from file
+            inDirPart = ''
+            if cfgGen.getboolean('initialiseParticlesFromFile'):
+                if cfgGen['particleFile']:
+                    inDirPart = cfgGen['particleFile']
+                else:
+                    partDirName = logName
+                    inDirPart = os.path.join(cfgGen['avalancheDir'], 'Outputs', 'com1DFA', 'particles', partDirName)
+            cfg['GENERAL']['inDirPart'] = inDirPart
```

Comment:
Why is this not in the initializeSimulation? Since thats where the particles are initialized... And initializeParticles also has the flag checking in it?

And I think it should only be:
```
if Particles from file:
   Read from file
```
and if the particles produced by com1DFA are wanted, then this is specified in `particleFile` (which can easily done at runtime without the need to touch the ini-File)

Reply:
Not sure I understand exactly what you want...